library/src/main/java/org/xmtp/android/library/Client.kt

```diff
@@ -137,12 +155,14 @@ class Client() {
         address: String,
         privateKeyBundleV1: PrivateKeyBundleV1,
         apiClient: ApiClient,
+        libXmtpClient: FfiXmtpClient? = null,
     ) : this() {
-        this.address = address
+        this.address = libXMTPClient?.accountAddress() ?: address
```

Comment:
I'd suggest just using whatever address you already are using for the client address. The one from `libxmtp` is going to be lowercased and the existing one might be checksum cased, so I wouldn't want to cause any V2 issues.